integrated_channels/cornerstone/exporters/learner_data.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""
+Learner data exporter for Enterprise Integrated Channel Cornerstone.
+"""
+
+
+from __future__ import absolute_import, unicode_literals
+
+from logging import getLogger
+
+from django.apps import apps
+
+from enterprise.api_client.discovery import get_course_catalog_api_service_client
+from integrated_channels.integrated_channel.exporters.learner_data import LearnerExporter
+from integrated_channels.utils import parse_datetime_to_epoch_millis
+
+LOGGER = getLogger(__name__)
+
+
+class CornerstoneLearnerExporter(LearnerExporter):
+    """
+    Class to provide a Cornerstone learner data transmission audit prepared for serialization.
+    """
+
+    def get_learner_data_records(self, enterprise_enrollment, completed_date=None, grade=None, is_passing=False):
+        """
+        Return a CornerstoneLearnerDataTransmissionAudit with the given enrollment and course completion data.
+
+        If completed_date is None, then course completion has not been met.
+
+        CornerstoneLearnerDataTransmissionAudit object should exit already if not then return None.
+        """
+        completed_timestamp = None
+        course_completed = False
+        if completed_date is not None:
+            completed_timestamp = parse_datetime_to_epoch_millis(completed_date)
+            course_completed = True
+
+        CornerstoneLearnerDataTransmissionAudit = apps.get_model(  # pylint: disable=invalid-name
+            'cornerstone',
+            'CornerstoneLearnerDataTransmissionAudit'
+        )
+
+        course_catalog_client = get_course_catalog_api_service_client(
+            site=enterprise_enrollment.enterprise_customer_user.enterprise_customer.site
+        )
+        try:
+            csod_learner_data_transmission = CornerstoneLearnerDataTransmissionAudit.objects.get(
+                user_id=enterprise_enrollment.enterprise_customer_user.user.id,
+                course_id=course_catalog_client.get_course_id(enterprise_enrollment.course_id),
```

Comment:
Because `enterprise_enrollment.course_id` is a course_run_id, on the other hand, `CornerstoneLearnerDataTransmissionAudit` accepts course key. So,`course_catalog_client.get_course_id` converts course_run_id into course_key/course_id.